service/src/main/resources/db/migration/V446__pedagogical_document_modified_by.sql

```diff
@@ -0,0 +1,14 @@
+ALTER TABLE pedagogical_document
+    ADD COLUMN modified_at TIMESTAMP WITH TIME ZONE,
+    ADD COLUMN modified_by UUID REFERENCES evaka_user;
+
+UPDATE pedagogical_document SET modified_at = created;
+UPDATE pedagogical_document SET modified_by = created_by;
+
+UPDATE pedagogical_document SET modified_at = updated;
+UPDATE pedagogical_document SET modified_by = updated_by;
```

Comment:
Or actually.. we should just rename `updated_by` to `modified_by`. While at it we could also rename `updated` to `updated_at` and remove it from known violations in `SchemaConventionsTest.kt`. (trigger needs to be recreated too if you do that)

We need both `updated_at` and `modified_at` because the first is updated by trigger and the second manually, so their semantics are different, but the `updated_by` is not needed separately from `modified_by`.

Reply:
The idea was that modified_at and modified_by would get the value of the updated or updated_by column if that was not null, or otherwise the value of the created or created_by column — allowing the new columns to be `NOT NULL`, and following the assumption made elsewhere that creating the row counts as a modification. I suppose an argument can be made for making the semantics of modified_at be the same as updated_at, though, maybe?

I also noticed that we are already bringing the value of `created` (which we could rename to `created_at` to follow the convention) up to the UI, although the card implies that that shouldn't happen. I assumed there was a reason that was already done that way, and so added a new last modified column rather than replacing the "Date" (`created`) column in the UI, but it would be easy enough to do that way as well. 

You are in any case right that this can be rewritten as one statement.

Reply:
Hmm, not sure if I fully understood, maybe quicker to chat this through after a daily. 

If you first do 
`UPDATE pedagogical_document SET modified_at = created;` 
and then 
`UPDATE pedagogical_document SET modified_at = updated;`
doesn't it always end up with the value of `updated`, even if it is `null`? I could be wrong but AFAIK the first statement does not do anything because it always gets overwritten in the second update.

Reply:
No, you're right, and now I'm not sure what possessed me to try to do it like that — I probably envisioned having having a WHERE clause and then just somehow never typed it in. I guess even natural intelligence can hallucinate.

Either way it would be better to use coalesce.